### PR TITLE
Reworded the explanation about when a lock is released

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.6+ |
| Fixed tickets | #4877 |

### CODE REVIEW

**Comment by xabbuh** [comment]
What about using a `caution` directive for this?

**Comment by javiereguiluz** [comment]
I'm usually against admonitions because in my opinion they break the reading flow and they're very annoying. But if other people agree with @xabbuh's suggestion, I'll do it.

File: components/filesystem/lock_handler.rst [hunk 1]
```diff
@@ -62,8 +62,23 @@ You can pass an optional blocking argument as the first argument to the
...
-The resource is automatically released by PHP at the end of the script. In
-addition, you can invoke the
-:method:`Symfony\\Component\\Filesystem\\LockHandler::release` method to release
-the lock explicitly. Once it's released, any other process can lock the
-resource.
+Beware that the resource lock is automatically released as soon as PHP applies the
+garbage-collection process to the ``LockHandler`` object. This means that if you
+refactor the first example showed in this article as follows:
+
+.. code-block:: php
```
Comment (WouterJ) [inline components/filesystem/lock_handler.rst]: use the shorthand `::` syntax

File: components/filesystem/lock_handler.rst [hunk 2]
```diff
@@ -62,8 +62,23 @@ You can pass an optional blocking argument as the first argument to the
...
+
+    use Symfony\Component\Filesystem\LockHandler;
+
+     if (!(new LockHandler('hello.lock'))->lock()) {
```
Comment (WouterJ) [inline components/filesystem/lock_handler.rst]: there is one space too much in the indent of this line

Comment (WouterJ) [inline components/filesystem/lock_handler.rst]: also, I don't like this new syntax. It's too much stuff in one line and works only in >=PHP 5.4

**Comment by WouterJ** [comment]
I like a caution too

File: components/filesystem/lock_handler.rst (hunk 2 above)
Comment (javiereguiluz) [inline components/filesystem/lock_handler.rst]: @WouterJ it I'm right, this PHP 5.4 example is precisely needed to make the code fail as explained above.

**Comment by javiereguiluz** [comment]
@xabbuh @WouterJ as requested, I've added the new documentation inside a `caution` admonition.

**Comment by javiereguiluz** [comment]
In my opinion this PR can be considered *finished* and ready to be merged.

File: components/filesystem/lock_handler.rst [hunk 3]
```diff
@@ -62,8 +62,25 @@ You can pass an optional blocking argument as the first argument to the
...
 PHP code will wait indefinitely until the lock is released by another process.
 
-The resource is automatically released by PHP at the end of the script. In
-addition, you can invoke the
-:method:`Symfony\\Component\\Filesystem\\LockHandler::release` method to release
-the lock explicitly. Once it's released, any other process can lock the
-resource.
+.. caution::
+
+    Beware that the resource lock is automatically released as soon as PHP applies
```
Comment (WouterJ) [inline components/filesystem/lock_handler.rst]: "Beware" should be used with "of". "Beware of the fact that the resource lock is automatically released"

Furthermore, I think "Be aware" is better in this context.

File: components/filesystem/lock_handler.rst [hunk 4]
```diff
@@ -62,8 +62,25 @@ You can pass an optional blocking argument as the first argument to the
...
+    the garbage-collection process to the ``LockHandler`` object. This means that
+    if you refactor the first example showed in this article as follows:
+
+    .. code-block:: php
```
Comment (WouterJ) [inline components/filesystem/lock_handler.rst]: use shorthand php

**Comment by javiereguiluz** [comment]
@WouterJ I've fixed the errors that you spotted. Thanks.

File: components/filesystem/lock_handler.rst [hunk 5]
```diff
@@ -62,8 +62,24 @@ You can pass an optional blocking argument as the first argument to the
...
-The resource is automatically released by PHP at the end of the script. In
-addition, you can invoke the
-:method:`Symfony\\Component\\Filesystem\\LockHandler::release` method to release
-the lock explicitly. Once it's released, any other process can lock the
-resource.
+.. caution::
+
+    Be aware of the fact that the resource lock is automatically released as soon
+    as PHP applies the garbage-collection process to the ``LockHandler`` object.
+    This means that if you refactor the first example showed in this article as
```
Comment (xabbuh) [inline components/filesystem/lock_handler.rst]: shown

File: components/filesystem/lock_handler.rst [hunk 6]
```diff
@@ -62,8 +62,24 @@ You can pass an optional blocking argument as the first argument to the
...
+
+        use Symfony\Component\Filesystem\LockHandler;
+
+         if (!(new LockHandler('hello.lock'))->lock()) {
+            // the resource "hello" is already locked by another process
+
+            return 0;
+        }
+
+    Now the code won't work as expected, because PHP's garbage collection mechanism
```
Comment (xabbuh) [inline components/filesystem/lock_handler.rst]: Do we really need a comma before "because"?

**Comment by WouterJ** [comment]
Labelled it as finished

**Comment by weaverryan** [comment]
Thanks Javier!  This is actually a really interesting PR. Cheers!

**Pull request merged**